threads-api/src/threads-api.ts

```diff
@@ -1141,3 +1206,14 @@ export class ThreadsAPI {
     }
   };
 }
+
+type Simplify<T> = {} & { [P in keyof T]: T[P] };
+
+type StrictUnion<T> = CombineUnion<T> extends infer U
+  ? T extends any
+    ? Simplify<T & { [K in Exclude<keyof U, keyof T>]?: undefined }>
+    : never
+  : never;
+
+type CombineUnion<T> = Pick<T, keyof T> &
+  Partial<(T extends any ? (x: T) => any : never) extends (x: infer U) => any ? U : never>;
```

Comment:
Have a feeling that we'll use this handy! Can we move this to `types/utils.ts` or some other separate file besides `threads-api.ts`?